contentful/apps
# feat: [INTEG-1789] - Add Vercel sidebar deploy button

## Purpose
The purpose of this update is to empower content editors with the ability to deploy configured instances to the production target directly from the Contentful space using the sidebar in our custom app. This change addresses the need for a streamlined deployment process, enhancing the user experience and efficiency in managing content deployments.

Ticket: [INTEG-1789](https://contentful.atlassian.net/browse/INTEG-1789)

## Approach
We chose an approach that integrates deployment functionality seamlessly into the content editing workflow. By enabling deployment configurations within the Contentful space and incorporating a dedicated sidebar in our custom app, we aim to provide a user-friendly and cohesive experience.

## Testing Steps

1. Ensure app is configured with a valid access token, follow instructions for creating one if not. 
<img width="1661" alt="Screenshot 2024-03-08 at 8 01 00 AM" src="https://github.com/contentful/apps/assets/22120469/55df25c4-c8f1-4d0b-9b2b-cc7f9fbd874a">


2. Open an entry for configured content type, and observe deploy button at the top of the sidebar for configured content type.
<img width="1418" alt="Screenshot 2024-03-08 at 8 01 25 AM" src="https://github.com/contentful/apps/assets/22120469/b05d3a9b-f7e2-4966-ba10-139a800a1cb0">


3. After initiating build, verify connected/configured project is queued and/or building
<img width="1222" alt="Screenshot 2024-03-08 at 8 01 46 AM" src="https://github.com/contentful/apps/assets/22120469/a4e21837-703a-4391-89be-befcdb4ae5a7">

## Considerations

Currently, Content Type selection persists to installation parameters but has no meaningful impact on user journey, this is pre-release functionality to lay foundation for deployments while features continued to be flushed out.

App configuration doesn't fully respect the comps while we continue to flush out features. Domain configuration is still under discovery, as well as live mode.


[INTEG-1789]: https://contentful.atlassian.net/browse/INTEG-1789?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
